bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastAudioSink.java

```diff
@@ -30,52 +37,103 @@
     private static final String MIME_TYPE_AUDIO_MPEG = "audio/mpeg";
 
-    private final ChromecastCommander commander;
+    private final ChromecastHandler handler;
     private final AudioHTTPServer audioHTTPServer;
     private final @Nullable String callbackUrl;
 
-    public ChromecastAudioSink(ChromecastCommander commander, AudioHTTPServer audioHTTPServer,
+    public ChromecastAudioSink(ChromecastHandler handler, AudioHTTPServer audioHTTPServer,
             @Nullable String callbackUrl) {
-        this.commander = commander;
+        this.handler = handler;
         this.audioHTTPServer = audioHTTPServer;
         this.callbackUrl = callbackUrl;
     }
 
-    public void process(@Nullable AudioStream audioStream) throws UnsupportedAudioFormatException {
+    @Override
+    public String getId() {
+        return handler.getThing().getUID().toString();
+    }
+
+    @Override
+    public @Nullable String getLabel(@Nullable Locale locale) {
+        return handler.getThing().getLabel();
+    }
+
+    @Override
+    public void processAsynchronously(@Nullable AudioStream audioStream) throws UnsupportedAudioFormatException {
         if (audioStream == null) {
             // in case the audioStream is null, this should be interpreted as a request to end any currently playing
             // stream.
             logger.trace("Stop currently playing stream.");
-            commander.handleCloseApp(OnOffType.ON);
+            handler.stop();
         } else {
             final String url;
             if (audioStream instanceof URLAudioStream) {
                 // it is an external URL, the speaker can access it itself and play it.
                 URLAudioStream urlAudioStream = (URLAudioStream) audioStream;
                 url = urlAudioStream.getURL();
+                tryClose(audioStream);
             } else {
                 if (callbackUrl != null) {
                     // we serve it on our own HTTP server
                     String relativeUrl;
-                    if (audioStream instanceof FixedLengthAudioStream) {
-                        relativeUrl = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, 10);
-                    } else {
-                        relativeUrl = audioHTTPServer.serve(audioStream);
+                    try {
+                        StreamServed streamServed = audioHTTPServer.serve(audioStream, 10, true);
+                        relativeUrl = streamServed.url();
+                        // we have to run the delayed task when the server has completely played the stream
+                        streamServed.playEnd().thenRun(() -> this.playbackFinished(audioStream));
+                    } catch (IOException e) {
+                        logger.warn("Chromecast binding was not able to handle the audio stream (cache on disk failed)",
+                                e);
+                        tryClose(audioStream);
+                        return;
```

Comment:
In all other audio sinks, it was decided to throw an exception (UnsupportedAudioStreamException) in this case.
Can you please do it also here for consistency ?

Reply:
As the other point (calling handler.stop()) you mentioned is not on the table anymore, should I make another PR for this ?

Reply:
Yes, if possible, please.